tests/handlers/test_password_providers.py

```diff
@@ -266,8 +266,9 @@ def test_no_local_user_fallback_ui_auth(self):
         # first delete should give a 401
         channel = self._delete_device(tok1, "dev2")
         self.assertEqual(channel.code, 401)
-        # there are no valid flows here!
-        self.assertEqual(channel.json_body["flows"], [])
+        # m.login.password UIA is permitted because the auth provider allows it,
+        # even though the localdb does not.
+        self.assertEqual(channel.json_body["flows"], ["m.login.password"])
```

Comment:
this is actually me fixing a bug I introduced in #8835 :/

Reply:
This test seems to be failing actually?

Reply:
that's because I'm a silly.